cdap-runtime-ext-dataproc/src/main/java/io/cdap/cdap/runtime/spi/provisioner/dataproc/DataprocClient.java

```diff
@@ -648,7 +677,7 @@ void updateClusterLabels(String clusterName,
    * @param sshRuntimeMonitor {@code true} if SSH is used for runtime monitoring
    * @return {@code true} for pribvate IP only Dataproc cluster
    */
-  private boolean isInternalIPOnly(boolean privateInstance, boolean sshRuntimeMonitor) {
+  private boolean isInternalIPOnly(Network network, boolean privateInstance, boolean sshRuntimeMonitor) {
```

Comment:
Do we need to pass `network` explicitly? since it's a member variable and should already be available in `isInternalIpOnly` right?

Reply:
This change is intentional since the field can be `null`. So instead of checking for `null` everywhere, we only check it once in the createCluster method so that these methods call would never receive `null` for `network`.